enterprise/server/ociregistry/ociregistry.go

```diff
@@ -595,6 +620,58 @@ func writeManifestToAC(ctx context.Context, raw []byte, acClient repb.ActionCach
 	return cachetools.UploadActionResult(ctx, acClient, arRN, ar)
 }
 
+func fetchManifestFromAC(ctx context.Context, acClient repb.ActionCacheClient, ref gcrname.Reference, hash gcr.Hash) (*ocipb.OCIManifestContent, error) {
+	arKey := &ocipb.OCIActionResultKey{
+		Registry:      ref.Context().RegistryStr(),
+		Repository:    ref.Context().RepositoryStr(),
+		ResourceType:  ocipb.OCIResourceType_MANIFEST,
+		HashAlgorithm: hash.Algorithm,
+		HashHex:       hash.Hex,
+	}
+	arKeyBytes, err := proto.Marshal(arKey)
+	if err != nil {
+		updateCacheEventMetric(actionCacheLabel, missLabel)
```

Comment:
nit: do you want to do these metrics in the caller? seems like that would simplify things a bit

Reply:
I've opted for counting hits and misses for both the AC and the CAS. I could count overall cache hits and misses -- if we're able to fetch a manifest successfully, that counts as one hit (not multiple hits for the AC and the CAS).

I'd like to keep the fine granularity for now, to be able to spot any problems affecting only the AC or the CAS.